### PR TITLE
enh(UI) : refacto the Rich Text Editor

### PR DESCRIPTION
**Pull request by Med1233**
## Description

fix [issue](https://centreon.atlassian.net/browse/MON-23461)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/AlignPicker.tsx [hunk 1]
```diff
@@ -88,25 +88,25 @@ const AlignPicker = ({ disabled }: Props): JSX.Element => {
...
   return (
     <Menu>
-      <Menu.Button ariaLabel={elementFormat} disabled={disabled}>
-        {selectedFormat && (
-          <div className={classes.button}>
-            <selectedFormat.Icon /> {selectedFormat.label}
-          </div>
-        )}
+      <Menu.Button
+        ariaLabel="align picker"
```
Comment (Thebarda) [inline centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/AlignPicker.tsx:92]: Use variable

File: centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import OrderedListIcon from '@mui/icons-material/FormatListNumbered';
+
+import { Menu } from '../../../components';
+
+import { useStyles } from './ToolbarPlugin.styles';
+
+const options = [
+  {
+    Icon: UnorderedListIcon,
+    label: 'Unordered List',
```
Comment (Thebarda) [inline centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx:29]: Use variable

File: centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    value: 'bullet'
+  },
+  {
+    Icon: OrderedListIcon,
+    label: 'Ordered List',
```
Comment (Thebarda) [inline centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx:34]: Same comment

File: centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      </Menu.Button>
+      <Menu.Items className={classes.menuItems}>
+        <div className={classes.menu}>
+          {options.map(({ Icon, value, label }) => (
+            <Menu.Item
+              isActive={equals(value, elementList)}
+              key={value}
+              onClick={() => formatList(value)}
+            >
+              <Icon aria-label={label} />
```
Comment (Thebarda) [inline centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx:143]: Translate aria labels

**Review by Thebarda — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

File: centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx (hunk 4 above)
Comment (Med1233) [reply centreon/packages/ui/src/RichTextEditor/plugins/ToolbarPlugin/ListButton.tsx:143]: is it necessary to add a translation to the area label?

File: centreon/packages/ui/src/RichTextEditor/RichTextEditor.cypress.spec.tsx [hunk 5]
```diff
@@ -47,16 +48,30 @@ describe('Rich Text Editor', () => {
...
       cy.findByLabelText('RichTextEditor')
         .should('be.visible')
         .and('have.value', '');
 
       cy.findByLabelText('RichTextEditor').parent().contains('Type here...');
+
+      cy.makeSnapshot();
+    });
+
+    it('displays all format actions when the format button was clicked', () => {
```
Comment (bdauria) [inline centreon/packages/ui/src/RichTextEditor/RichTextEditor.cypress.spec.tsx:66]:
```suggestion
    it('displays all format actions when the format button is clicked', () => {
```

**Review by kduret — APPROVED** [review]
CODEOWNERS for cypress component tests will be fixed here : https://github.com/centreon/centreon/pull/2649

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**